checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsVisitor.java

```diff
@@ -0,0 +1,80 @@
+package org.checkerframework.checker.calledmethods;
+
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.StringJoiner;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.calledmethods.builder.BuilderFrameworkSupport;
+import org.checkerframework.checker.calledmethods.qual.CalledMethods;
+import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey;
+import org.checkerframework.common.accumulation.AccumulationVisitor;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.TreeUtils;
+
+/**
+ * This visitor implements the custom error message finalizer.invocation.invalid, and supports
+ * counting the number of framework build calls.
+ */
+public class CalledMethodsVisitor extends AccumulationVisitor {
+
+    /** Error message key for incorrect finalizer invocations. */
+    public static final @CompilerMessageKey String FINALIZER_INVOCATION_INVALID =
+            "finalizer.invocation.invalid";
```

Comment:
This is only used once.  There is no danger of misspelling because of the Compiler Message Checker.  So, please inline it to make the code shorter and easier to read.